_posts/2020-04-08-secure-software-development-lifecycle.md

```diff
@@ -0,0 +1,138 @@
+
+Software development is usually divided into different phases. There can be more or there can be fewer depending on the software project management framework in use. Phases can exists in short iterations or in a one long waterfall or anything between. These phases could be for example:
+
+![ssdlc phases](/img/ssdlc/phases.png)
+
+SSDLC adds security requirements in each of the phases. This makes you consider security in all phases of the software project. 
+
+## Why does secure software development lifecycle matter
+
+It is time for mandatorial motivation! Why does this matter? Why I am reading this? Is this yet another SAFe or ITIL? 
+
+You should think SSDLC to be more like agile development. Where agile development advocates for adaptiveness and continual improvement the SSDLC should do the same. By itself it does not mean anything unless you are able to implement a useful set of tools and processes that supports your business goals. The same applies for SSDLC. Main goal is to provide a set of security oriented best practices for your existing development practices. 
+
+![processes](/img/ssdlc/processes.jpg)
+
+So what happened to "build the software and let security engineers test it afterwards"? The same happened as with all the other testing too. It is much cheaper to catch mistakes early. Security problems in requirements and/or in design are pretty expensive to fix if you let them go to production. You want to fix them early. To catch them early you need supporting routines to have a systematic approach in finding them. That is what SSDLC is all about. 
+
+## How to implement secure software development lifecycle
+
+Start small and make it evolve over time. Treat the improvements in SSDLC like you would treat any other process improvements or quality assurance tasks. Make them as visible for in your agile boards and make a supporting document that clearly tells what you are after.
+
+#### Requirements
+
+There are laws and regulations. There are company policies and customer policies. Failing to notice these will make it difficult to deal with them later. Gathering intelligence on what you are about to do businesswise should make you able to also make some security requirements for the project. This will create the foundation of your security awareness level later on. Are you dealing with credit cards? Surely you have then heard about PCI-DSS. Are you creating an application for healthcare? Maybe you ought to know something about HIPAA. Know your industry and make use of whatever given requirements there are for the field. You should also have a hunch about data classification levels.
```

Comment:
The first two sentences should be rephrased with something like "Requirements stem from company policies.."